### PR TITLE
pre-processing data update

### PR DESCRIPTION
**Pull request by MikeLippincott**
quick PR that updates data pre-processing. Other PRs are needed for the whole repo to make senese. Breaking things into small PRs

### CODE REVIEW

File: 0.Preprocess_Data/nbconverted/0.preprocessing.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        + feature_df["Metadata_inducer2_concentration"].astype(str)
+    ).astype(str),
+]
+feature_df["Metadata_Dose"] = np.select(conditions, results)
+
+
+# In[12]:
+
+
+feature_df["Metadata_inducer1_concentration"] = pd.to_numeric(
```
Comment (jenna-tomkinson) [inline 0.Preprocess_Data/nbconverted/0.preprocessing.py:151]: This is just a recommendation, but maybe if you wanted to reduce the lines of code, you could add this to above code cell where you fillna. Example:

```python
feature_df["Metadata_inducer1_concentration"] = feature_df[
    "Metadata_inducer1_concentration"
].fillna(0).to_numeric()
```

File: 0.Preprocess_Data/nbconverted/0.preprocessing.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    feature_df["Metadata_inducer1_concentration"]
+)
+
+
+# ## N Beta Column condition generation
+
+# In[13]:
+
+
+feature_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"] = (
```
Comment (jenna-tomkinson) [inline 0.Preprocess_Data/nbconverted/0.preprocessing.py:161]: I would recommend adding code comments above each of these columns you are generating with more info on what they are/how they are different.

File: 0.Preprocess_Data/nbconverted/0.preprocessing.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+feature_df["threeb_Metadata_Treatment_Dose_Inhibitor_Dose"] = (
+    feature_df["Metadata_Treatment"]
+    + "__"
+    + feature_df["Metadata_Dose"].astype(str)
+    + "__"
+    + feature_df["Metadata_inhibitor"].astype(str)
+    + "_"
+    + feature_df["Metadata_inhibitor_concentration"].astype(str)
+).astype(str)
```
Comment (jenna-tomkinson) [inline 0.Preprocess_Data/nbconverted/0.preprocessing.py:189]:
```suggestion
).astype(str)

```
Recommend adding extra space between the columns to stay consistent with the above spacing.

File: 0.Preprocess_Data/nbconverted/0.preprocessing.py [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+    + feature_df["Metadata_inhibitor"].astype(str)
+    + "__"
+    + feature_df["Metadata_inhibitor_concentration"].astype(str)
+).astype(str)
+
+
+# In[ ]:
+
+
+# feature_df.to_csv(feature_df_out_path, index=False)
```
Comment (jenna-tomkinson) [inline 0.Preprocess_Data/nbconverted/0.preprocessing.py:204]: Same as above, I recommend removed unused code/commented-out code.

**Review by jenna-tomkinson — COMMENTED** [review]
Great PR! I have left some general questions and some recommendations. I did notice that there were a lot of imports in this file that weren't used, so that is why I am leaving a comment and not approving. 

Once those are imports are removed and comments addressed, it LGTM!

File: 0.Preprocess_Data/nbconverted/0.preprocessing.py [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+
+# In[2]:
+
+
+# Define inputs
+feature_file = pathlib.Path(
+    "../../Extracted_Features_(CSV_files)/SHSY5Y_sc_norm.parquet"
+)
+feature_df = pq.read_table(feature_file).to_pandas()
+# feature_df = pd.read_csv(feature_file, engine="pyarrow")
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 0.Preprocess_Data/nbconverted/0.preprocessing.py:37]: removed

File: 0.Preprocess_Data/nbconverted/0.preprocessing.py [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+import sys
+
+import numpy as np
+import pandas as pd
+import pdfkit
+import plotly.express as px
+import pyarrow as pa
+import pyarrow.parquet as pq
+import seaborn as sns
+from matplotlib import pyplot as plt
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 0.Preprocess_Data/nbconverted/0.preprocessing.py:17]: removed

File: 0.Preprocess_Data/nbconverted/0.preprocessing.py [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+from pycytominer.cyto_utils import infer_cp_features
+from sklearn.linear_model import LinearRegression
+from sklearn.preprocessing import LabelEncoder
+
+# import Union
+
+
+sys.path.append("..")
+# from ..utils.utils import df_stats
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 0.Preprocess_Data/nbconverted/0.preprocessing.py:26]: removed

File: 0.Preprocess_Data/nbconverted/0.preprocessing.py [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+import matplotlib.pyplot as plt
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 0.Preprocess_Data/nbconverted/0.preprocessing.py:27]: removed

File: 0.Preprocess_Data/nbconverted/0.preprocessing.py [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+import pathlib
+import sys
+
+import numpy as np
+import pandas as pd
+import pdfkit
+import plotly.express as px
+import pyarrow as pa
+import pyarrow.parquet as pq
+import seaborn as sns
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 0.Preprocess_Data/nbconverted/0.preprocessing.py:16]: removed

File: 0.Preprocess_Data/nbconverted/0.preprocessing.py [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+# In[1]:
+
+
+import pathlib
+import sys
+
+import numpy as np
+import pandas as pd
+import pdfkit
+import plotly.express as px
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 0.Preprocess_Data/nbconverted/0.preprocessing.py:13]: removed

File: 0.Preprocess_Data/nbconverted/0.preprocessing.py [hunk 11]
```diff
@@ -0,0 +1,206 @@
...
+
+# In[1]:
+
+
+import pathlib
+import sys
+
+import numpy as np
+import pandas as pd
+import pdfkit
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 0.Preprocess_Data/nbconverted/0.preprocessing.py:12]: removed

File: 0.Preprocess_Data/nbconverted/0.preprocessing.py [hunk 12]
```diff
@@ -0,0 +1,206 @@
...
+
+print(feature_df.shape)
+feature_df.head()
+
+
+# In[5]:
+
+
+# removing costes features as they behave with great variance across all data
+feature_df = feature_df.drop(feature_df.filter(regex="Costes").columns, axis=1)
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 0.Preprocess_Data/nbconverted/0.preprocessing.py:60]: yes! Confirmed

File: 0.Preprocess_Data/nbconverted/0.preprocessing.py [hunk 13]
```diff
@@ -0,0 +1,206 @@
...
+
+# In[7]:
+
+
+# Recycled code from: https://github.com/WayScience/NF1_SchwannCell_data/blob/main/5_analyze_data/notebooks/linear_model/fit_linear_model.ipynb
+cell_count_df = (
+    feature_df.groupby("Metadata_Well")["Metadata_Plate"]
+    .count()
+    .reset_index()
+    .rename(columns={"Metadata_Plate": "Metadata_number_of_singlecells"})
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 0.Preprocess_Data/nbconverted/0.preprocessing.py:80]: this is a code hangover. I am deleting. Needed this when single cells were not added to df from wave3 plate1.

File: 0.Preprocess_Data/nbconverted/0.preprocessing.py [hunk 14]
```diff
@@ -0,0 +1,206 @@
...
+
+
+feature_df.head()
+
+
+# In[9]:
+
+
+# show max column in pandas df
+pd.set_option("display.max_columns", 100)
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 0.Preprocess_Data/nbconverted/0.preprocessing.py:94]: removed. I was using this for confirming dataframe structures manually

File: 0.Preprocess_Data/nbconverted/0.preprocessing.py [hunk 15]
```diff
@@ -0,0 +1,206 @@
...
+].fillna(0)
+
+
+# #### Combine Inducer1 and Inducer2 into one column
+
+# In[11]:
+
+
+# treatment column merge
+conditions = [
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 0.Preprocess_Data/nbconverted/0.preprocessing.py:118]: this is combining inducer 1 and inducer 2 but I need this as combining string with null returns null

File: 0.Preprocess_Data/nbconverted/0.preprocessing.py (hunk 2 above)
Comment (MikeLippincott) [reply 0.Preprocess_Data/nbconverted/0.preprocessing.py:161]: added documentation

File: 0.Preprocess_Data/nbconverted/0.preprocessing.py (hunk 4 above)
Comment (MikeLippincott) [reply 0.Preprocess_Data/nbconverted/0.preprocessing.py:205]: done

**Review by jenna-tomkinson — APPROVED** [review]
LGTM! Great job!

**Pull request merged**